web/src/lib/utils/asset-utils.ts

```diff
@@ -438,6 +439,32 @@ export const deleteStack = async (stackIds: string[]) => {
   }
 };
 
+export const keepThisDeleteOthers = async (keepId: string, stackId: string) => {
+  const $t = get(t);
+
+  try {
+    const stack = await getStack({ id: stackId });
+    const assetToKeep = stack.assets.find((asset) => asset.id === keepId);
```

Comment:
You can remove these after changing the function signature.

Reply:
Commit has been pushed to address this.